OPHD/States/CrimeExecution.cpp

```diff
@@ -0,0 +1,52 @@
+#include "CrimeExecution.h"
+#include "../StructureManager.h"
+#include <NAS2D/StringUtils.h>
+#include <NAS2D/Utility.h>
+#include <algorithm>
+
+
+CrimeExecution::CrimeExecution(NotificationArea& notificationArea) : mNotificationArea(notificationArea) {}
+
+
+void CrimeExecution::executeCrimes(std::vector<Structure*> structuresCommittingCrime)
+{
+	for (auto& structure : structuresCommittingCrime)
+	{
+		if (structure == nullptr) 
+		{
+			continue;
+		}
+
+		switch (structure->structureId())
+		{
+		case StructureID::SID_AGRIDOME:
+			executeStealingFood(static_cast<FoodProduction&>(*structure));
+			break;
+		}
+	}
+}
+
+
+void CrimeExecution::executeCrimes(std::vector<Structure*>&& structuresCommittingCrime)
+{
+	// Delegate to lvalue function overload
+	executeCrimes(structuresCommittingCrime);
+}
```

Comment:
Hmm, this binds to temporaries, but then forces a copy of the temporary when passing it to the by-value overload. Could eliminated this overload get the same effect.

Though I assume this wasn't what was intended.